fairscale/nn/misc/grad_bucket.py

```diff
@@ -70,11 +71,15 @@ def add_grad(self, param: torch.Tensor) -> None:
         """
         Add a new parameter gradient to the bucket. Param.grad becomes a view of this bucket buffer
         """
+
+        assert id(param) not in self._param_ids, "The same gradients cannot be checked in twice"
```

Comment:
when a module is forwarded multiple times, like the multi_res_input_forward in vissl, perhaps this will happen?

Reply:
it's defined at construction time, not per forward, so this should be fine. I think that the assert is important because the grad becomes a view of a tensor, and it cannot be a (correct) view of two things at the same time